### PR TITLE
Add Audiencevalidator

### PR DESCRIPTION
**Pull request by mwdb**

### CODE REVIEW

File: spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java [hunk 1]
```diff
@@ -623,13 +618,22 @@ private URI replaceSubdomain(URI uri, String subdomain) {
...
 
+	@Override
+	public String toString() {
+		try {
+			if (GRANTTYPE_CLIENTCREDENTIAL.equals(getGrantType())) {
+				return String.format("%s (client)", getClientId());
+			}
+			else
+			{
+				return String.format("%s / %s (user)",getLogonName(), getOrigin());
```
Comment (nenaraab) [inline spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java]: @mwdb 
Unfortunately your pull request does not fix the issue. It must be some character sequence, that can be used as database entry for the Spring ACL tables, e.g. `getLogonName()`.

Comment (nenaraab) [reply spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java]: please have a look into my pull request:
https://github.com/SAP/cloud-security-xsuaa-integration/pull/10

Comment (mwdb) [reply spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java]: merged your pr

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/AudienceValidator.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	 * @param token
+	 * @return list of audiences
+	 */
+	private List<String> allowedAudiences(Jwt token) {
+		List<String> allAudiences = new ArrayList<>();
+		if (token.getClaimAsString("aud") != null) {
+			allAudiences.addAll(token.getClaimAsStringList("aud"));
+		}
+
+		if (allAudiences.size() == 0 && token.getClaimAsStringList("scope").size()>0) {
```
[the remark this answers is not in the document]
Comment (mwdb) [reply spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/authentication/AudienceValidator.java]: real worlsd tokens always have opeenid or uaa.resource

File: spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java (hunk 1 above)
Comment (nenaraab) [reply spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java]: thanks!

**Pull request merged**